src/utils/thread_helper.cpp

```diff
@@ -1360,8 +1392,48 @@ otError ThreadHelper::RetrieveTelemetryData(Mdns::Publisher *aPublisher, threadn
         }
         // End of Nat64Mapping section.
 #endif // OTBR_ENABLE_NAT64
+#if OTBR_ENABLE_DHCP6_PD
+        // Start of Dhcp6PdState section.
+        {
+            wpanBorderRouter->set_dhcp6_pd_state(
+                Dhcp6PdStateFromOtDhcp6PdState(otBorderRoutingDhcp6PdGetState(mInstance)));
+        }
+        // End of Dhcp6PdState section.
 
-        // End of WpanBorderRouter section.
+        // Start of PD prefix
+        {
+            otBorderRoutingPrefixTableEntry aPrefixInfo;
+            Sha256::Hash                    hash;
+            Sha256                          sha256;
+
+            otBorderRoutingGetPdOmrPrefix(mInstance, &aPrefixInfo);
+            uint8_t pdPrefixShaInput[OT_IP6_ADDRESS_SIZE + kNat64PdCommonHashSaltLength];
+            memcpy(pdPrefixShaInput, aPrefixInfo.mPrefix.mPrefix.mFields.m8,
+                   sizeof(aPrefixInfo.mPrefix.mPrefix.mFields.m8));
+            memcpy(&pdPrefixShaInput[sizeof(aPrefixInfo.mPrefix.mPrefix.mFields.m8)], mNat64PdCommonSalt,
+                   sizeof(mNat64PdCommonSalt));
+
+            sha256.Start();
+            sha256.Update(pdPrefixShaInput, sizeof(pdPrefixShaInput));
+            sha256.Finish(hash);
+
+            wpanBorderRouter->mutable_hashed_pd_prefix()->append(reinterpret_cast<const char *>(hash.GetBytes()),
+                                                                 sizeof(hash.GetBytes()));
+        }
+        // End of PD prefix
+        // Start of DHCPv6 PD processed RA Info
```

Comment:
// Start of DHCPv6 PdProcessedRaInfo section.
Use same name as in proto may help code search slightly easier.

Reply:
Thanks!